lib/AST/NameLookup.cpp

```diff
@@ -1499,9 +1499,13 @@ bool DeclContext::lookupQualified(Type type,
     if (tracker)
       tracker->addUsedMember({current, member.getBaseName()},isLookupCascading);
 
-    // Make sure we've resolved implicit constructors, if we need them.
-    if (member.getBaseName() == ctx.Id_init && typeResolver)
-      typeResolver->resolveImplicitConstructors(current);
+    // Make sure we've resolved implicit members, if we need them.
+    if (typeResolver) {
+      if (member.getBaseName() == ctx.Id_init)
+        typeResolver->resolveImplicitConstructors(current);
+
+      typeResolver->resolveImplicitMember(current, member);
```

Comment:
@jrose-apple So, it seems that it's possible to end up doing qualified lookups through performing an `UnqualifiedLookup` in `resolveWitnessViaLookup`. However, I'm having a hard time constructing a scenario in which it would be possible to enter a loop — since we only do lookups upwards/outwards (e.g. from nested types out), I don't think we could enter a loop since we can't have inheritance loops and things like that. Thoughts?